### ISSUE TITLE
Can't use the keyboard shortcut to open the command palette on Firefox

### ISSUE THREAD

**Issue by vharsh** [jupyter/notebook#2645]
Firefox uses `[ctrl] + [shift] + [p]` to open a Private Window 
and Chromium uses it to open the print dialog when I am *not* in insert mode.
I have not been able to open the palette in Firefox when I am in any mode, I can open it in Chromium when I am in Insert mode.

Should it be changed to something else which isn't already a shortcut in the popular browsers, so that it can be opened irrespective of the mode the user is in?

**Comment by takluyver** [comment]
We already added Ctrl-Shift-F as an alternative shortcut for the same thing.

**Comment by vharsh** [comment]
Sorry for this issue. This wasn't listed in the docs.

**Comment by takluyver** [comment]
Want to make a PR for the docs? ;-)

**Comment by vharsh** [comment]
I'd love to after I find a few similar things to fix. 
I had found a typo of toolbar, I can't find it right now.
Moreover I have some confusions between server-extensions and nbserver extensions, regarding which one is deprecated and which one is more suitable for a particular use-case.

**Comment by takluyver** [comment]
`nbserver_extensions` is recommended now, `server_extensions` is deprecated.

**Comment by vharsh** [comment]
I have a little confusion regarding this. 
What I know as of now, by reading...
* Official Jupyter [extending docs](http://jupyter-notebook.readthedocs.io/en/latest/extending/)
    - Javascript API is **unstable** and **will break** the next [release](http://jupyter-notebook.readthedocs.io/en/latest/extending/frontend_extensions.html) 
My conclusion : This is unstable probably not deprecated. Should work in 5.0 and maybe the extension needs tweaks for Notebook 5.1 . This should be the optimal way to add buttons to dashboard using Jupyter.notebook object as shown.

* [recommended blog](https://mindtrove.info/4-ways-to-extend-jupyter-notebook/)
   - Point#2 This one looks deprecated.
   - Point#3 Jupyter Notebook extensions (nbextensions) same as the one listed above, has the `nb` in it, probably same as `nbserver_extensions`
   - Point#4 Notebook Server Extensions Is this one going to be deprecated or is this `nbserver_extension`

The docs probably do not clearly specify as of now which one of it is which.
Please tell me if I am moving in the right direction. Thanks a lot. 

- vharsh on Freenode

**Comment by takluyver** [comment]
Yeah, the docs have basically always said that the Javascript API isn't stable. It hasn't stopped people using it. ;-)

From that blog post:

* IPython kernel extensions are still around as well, not deprecated. Docs here: http://ipython.readthedocs.io/en/stable/config/extensions/index.html
* nbextensions means notebook frontend javascript extensions
* server_extensions and nbserver_extensions both specify the same thing in different ways. nbserver_extensions, which is a dict, is the recommended way. These are Python modules which extend the notebook server process, not the frontend javascript.

> Add some functionality to make interaction with git&github easier from the front-end. I am not getting very helpful messages in the browser's JavaScript console, which points to some main.min.js.

Yeah, sorry about that. There are supposed to be source maps which let it point to the real location, but they don't seem to work reliably.

> I can bring up buttons by writing JavaScript in the Notebook cell with the %%javascript magic-flags. I can't figure out if the .js file has to be in ~/.jupyer/custom/custom.js or ~/.local/share/jupyter/nbextensions.

`custom.js` is intended to be a simple mechanism for a user to customise their own installation. Nbextensions are for extension packages which you can write and distribute. Nbextensions have to be enabled (see `jupyter nbextension enable --help`) before they are loaded, whereas `custom.js` is loaded by default if it exists.
